### PR TITLE
YAML and JSON are scanned like structures

### PR DESCRIPTION
**Pull request by babenek**
## Description

- Added StructContentProvider to separate structure storage
- YAML and JSON might be scanned like structures
- The case in yaml with multiline credential is found:
```
key: |
  5UpEr5eCrEt
```

## How has this been tested?

- [x] UnitTests
- [x] do benchmark with --dept 33

### CODE REVIEW

**Comment by csh519** [comment]
@babenek 

Okay I understood what you mean except about whatthepatch case(xDizzix)..
Is it related with recursive scan with proper file type(extension)?
And regarding propagation of `config`, we can make a `config` as a `classmethod` and make it access from all over source code.

But I love singleton style more as we using.

File: credsweeper/file_handler/struct_content_provider.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+logger = logging.getLogger(__name__)
+
+
+class StructContentProvider(ContentProvider):
+    """Dummy raw provider to keep structured data
+
+    Parameters:
+        struct: byte sequence to be stored.
+        file_path: optional string. Might be specified if you know true file name lines was taken from.
```
Comment (kmnls) [inline credsweeper/file_handler/struct_content_provider.py:15]:
```suggestion
        file_path: optional string. Might be specified if you know true file name where lines were taken from.
```

**Review by kmnls — DISMISSED** [review]

File: credsweeper/file_handler/data_content_provider.py [hunk 2]
```diff
@@ -42,6 +45,47 @@ def data(self, data: bytes) -> None:
...
+            text = self.data.decode(encoding='utf-8', errors='strict')
+        except Exception:
+            return False
+        # JSON
+        try:
+            if "{" in text:
+                self.structure = json.loads(text)
+                logger.debug("CONVERTED from json")
+                # logger.debug("CONVERTED from '%s' json:\n%s", self.data.decode(encoding='utf-8', errors='strict'),
+                #              str(self.structure))
```
Comment (csh519) [inline credsweeper/file_handler/data_content_provider.py:62]: Could you please unify the logging style with `YAML` code block(L75-L76)?
And please remove annotated codes.

**Review by csh519 — DISMISSED** [review]
Thank you for adding structure scan.

LGTM :+1:

**Review by csh519 — DISMISSED** [review]
Approve again.

**Review by iuriimet — APPROVED** [review]
LGTM

**Pull request merged**